src/questions/QuestionDB.java

```diff
@@ -0,0 +1,345 @@
+package questions;
+
+import model.Question;
+import model.Category;
+
+/**
+ * @author Roger
+ */
+
+import java.sql.*;
+import java.util.ArrayList;
+
+public class QuestionDB implements QuestionStore {
+
+	private Connection connection = null;
+	private PreparedStatement preparedStatement = null;
+	private ResultSet resultSet = null;
+	
+
+	public QuestionDB(Connection connection) {
+
+		this.setConnection(connection);
+	}
+
+	public Connection getConnection() {
+		return connection;
+	}
+
+	private void setConnection(Connection connection) {
+		this.connection = connection;
+	}
+	
+	@Override
+	public Question getByID(int id) {
+
+		try {
+
+			String query = "SELECT questions.*, answers.a, answers.b, answers.c, answers.d, categories.name"
+					+ "FROM questions"
+					+ "LEFT JOIN answers ON questions.answer_id = answers.id"
+					+ "LEFT JOIN categories ON questions.category_id = categories.id"
+					+ "WHERE questions.id = " + id;
+			
+			preparedStatement = connection.prepareStatement(query);			
+			resultSet = preparedStatement.executeQuery();
+			
+			while (resultSet.next()) {
+				
+				//	Prepare answers
+				ArrayList<String> answers = new ArrayList<String>(); 
+				answers.add(resultSet.getString("answers.a"));
+				answers.add(resultSet.getString("answers.b"));
+				answers.add(resultSet.getString("answers.c"));
+				answers.add(resultSet.getString("answers.d"));
+				
+				//	Prepare category
+				Category category = null;
```

Comment:
Die Frage ist exemplarisch auch für die anderen Methoden. Wenn ich jetzt aus dem JOIN Query meine Antworten gezogen habe kann ich die dann in die ArrayList übergeben. Wie mache ich das mit der Category. Da kann ich ja aus der Datenbank nur entweder die ID oder den Namen der Kategorie herausziehen. Wie speichere ich das dann als "Category"

Reply:
Ups! Das hab ich im Kontruktor übersehen ^^ Danke :D

Reply:
mh... Es scheint aber nicht so zu funktionieren wie wir uns das vorgestellt haben oder? :X 
Alias types gibt es in Java scheinbar auch nicht. Es wäre zwar etwas overkill, aber wir könnten aus der Category eine normale Klasse machen. 

Google heute abend mal was es noch für alternativen gehen würde :x:

Reply:
Hab es zur Klasse gemacht :)